### ISSUE TITLE
finetune() function in validation.py  seems a little bit strange

### ISSUE THREAD

**Issue by vampytam** [ZhijiangTang/ECG-Benchmark#2]
the training metric calculation is using the batch_Y from training data_loader and out embedding from test_loader:

https://github.com/ZhijiangTang/ECG-Benchmark/blob/a95df00badbbfcddc0221f3847a24ae85fb3177a/validation/validation.py#L103C1-L106C22

https://github.com/ZhijiangTang/ECG-Benchmark/blob/a95df00badbbfcddc0221f3847a24ae85fb3177a/models/finetune_model.py#L105

Could the codes be fine-tuned a little bit more :)

and If possible, could you guys provide the codes of data pre-processing and pretraining other models🙏 

Thank you sincerely
